vector/src/main/java/im/vector/riotx/features/home/room/list/RoomListViewModel.kt

```diff
@@ -89,10 +90,21 @@ class RoomListViewModel @AssistedInject constructor(@Assisted initialState: Room
         this.toggle(action.category)
     }
 
+    private fun handleFilter(action: RoomListActions.FilterWith) {
+        setState {
+            copy(
+                    roomFilter = action.filter
+            )
+        }
+    }
 
     private fun observeRoomSummaries() {
         homeRoomListObservableSource
                 .observe()
+                .observeOn(Schedulers.computation())
```

Comment:
Observe twice, no schedule on?

Reply:
observeOn() is used to change the thread